src/Components/Server/src/CircuitOptions.cs

```diff
@@ -45,5 +46,10 @@ public class CircuitOptions
         /// Defaults to <c>3 minutes</c>.
         /// </value>
         public TimeSpan DisconnectedCircuitRetentionPeriod { get; set; } = TimeSpan.FromMinutes(3);
+
+        /// <summary>
+        /// Gets the <see cref="IDataProtector"/> used to protect the generated circuit Ids.
+        /// </summary>
+        public IDataProtector CircuitIdProtector { get; set; }
```

Comment:
This is needed for crypto-agility

Reply:
Is the idea that a developer who wants to audit/standardize their use of crypto will be able to replace this with whatever they want?

Reply:
What are the expectations if this is set to null?

Reply:
Why not provide it via DI?

Reply:
Just follows the pattern used in other places across auth stuff. I believe cookies also do it this way.

Reply:
> What are the expectations if this is set to null?

The expectation is that you'll get a runtime exception. This is an advanced setting, so if you are changing this, the expectation is that you know what you are doing, so I'm not concerned about it.

Reply:
That's a legacy pattern leftover from Katana. Use DI.